core/modules/css/CssAccess.cc

```diff
@@ -0,0 +1,855 @@
+                                emptyChunkPath);
+}
+
+std::shared_ptr<CssAccess>
+CssAccess::makeKvCss(std::shared_ptr<KvInterface> const& kv, std::string const& emptyChunkPath) {
+    return std::shared_ptr<CssAccess>(new CssAccess(kv, std::make_shared<EmptyChunks>(emptyChunkPath)));
+}
+
+
+// Construct from KvInterface instance and empty chunk list instance
+CssAccess::CssAccess(std::shared_ptr<KvInterface> const& kvInterface,
+                     std::shared_ptr<EmptyChunks> const& emptyChunks,
+                     std::string const& prefix)
+    : _kvI(kvInterface), _emptyChunks(emptyChunks), _prefix(prefix) {
+}
+
+int
+CssAccess::cssVersion() {
+    return VERSION;
+}
+
+std::vector<std::string>
+CssAccess::getDbNames() const {
+    std::string p = _prefix + "/DBS";
+    return _kvI->getChildren(p);
+}
+
+bool
+CssAccess::containsDb(std::string const& dbName) const {
+    if (dbName.empty()) {
+        LOGF(_log, LOG_LVL_DEBUG, "Empty database name passed.");
+        return false;
+    }
+    std::string p = _prefix + "/DBS/" + dbName;
+    bool ret = _kvI->exists(p);
+    LOGF(_log, LOG_LVL_DEBUG, "containsDb(%1%): %2%" % dbName % ret);
+    return ret;
+}
+
+StripingParams
+CssAccess::getDbStriping(std::string const& dbName) const {
+    LOGF(_log, LOG_LVL_DEBUG, "getDbStriping(%1%)" % dbName);
+    StripingParams striping;
+    auto dbMap = _getSubkeys(_prefix + "/DBS/" + dbName, {"partitioningId"});
+    auto const& partId = dbMap["partitioningId"];
+    if (partId.empty()) {
+        // if database is not defined throw an exception, otherwise return default values
+        _assertDbExists(dbName);
+        return striping;
+    }
+
+    // get all keys
+    std::string pkey = _prefix + "/PARTITIONING/_" + partId;
+    std::vector<std::string> subKeys{"nStripes", "nSubStripes", "overlap"};
+    auto const keyMap = _getSubkeys(pkey, subKeys);
+
+    // fill the structure
+    try {
+        striping.partitioningId = std::stoi(partId);
+        auto iter = keyMap.find("nStripes");
+        if (iter != keyMap.end()) {
+            striping.stripes = std::stoi(iter->second);
+        }
+        iter = keyMap.find("nSubStripes");
+        if (iter != keyMap.end()) {
+            striping.subStripes = std::stoi(iter->second);
+        }
+        iter = keyMap.find("overlap");
+        if (iter != keyMap.end()) {
+            striping.overlap = std::stod(iter->second);
+        }
+    } catch (std::exception const& exc) {
+        throw KeyValueError(pkey, "one of the keys is not numeric: " + std::string(exc.what()));
+    }
+
+    return striping;
+}
+
+void
+CssAccess::createDb(std::string const& dbName,
+                    StripingParams const& striping,
+                    std::string const& storageClass,
+                    std::string const& releaseStatus) {
+    LOGF(_log, LOG_LVL_DEBUG, "createDb(%1%)" % dbName);
+
+    std::string partId;
+    if (striping.stripes > 0) {
+        // need to define partitioning structure
+
```

Comment:
Well, the comment says that we need to define that structure in case when `stripes > 0` and code in the if block actually does it, so we do define structure there.